src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -764,6 +770,34 @@ namespace levin
     );
   }
 
+  void notify::on_connection_new(const boost::uuids::uuid &id, bool is_income)
```

Comment:
Rename `on_handshake_complete` as that is equally important to the connection status, and refactor could easily lose this pertinent information.